python/pyarrow/types.pxi

```diff
@@ -697,6 +697,9 @@ cdef class ExtensionType(BaseExtensionType):
         else:
             return NotImplemented
 
+    def __str__(self):
```

Comment:
I think this can be removed now (after which the `str` test will need to be fixed).

Reply:
Just to confirm, if I remove the `__str__`, the output changes as follows. Is this what you're looking for? Thanks!

```
def test_ext_type_str():
    ty = IntegerType()
    expected = "extension<arrow.py_extension_type<IntegerType>>"
    assert str(ty) == expected
    assert pa.DataType.__str__(ty) == expected


def test_ext_type_repr():
    ty = IntegerType()
    expected = "IntegerType(extension<arrow.py_extension_type<IntegerType>>)"
    assert repr(ty) == expected
```

Reply:
The `str` looks fine, the `repr` looks a bit unfortunate (though not terrible). It seems the `repr` could be just the same as `str`.

Reply:
Okay, I've removed `__str__` and implemented `__repr__` instead to preserve `IntegerType(DataType(int64))` for `repr()`.